electricitymap/contrib/config/__init__.py

```diff
@@ -4,34 +4,77 @@
 
+import yaml
+
 ZoneKey = NewType("ZoneKey", str)
 Point = NewType("Point", Tuple[float, float])
 BoundingBox = NewType("BoundingBox", List[Point])
 
 CONFIG_DIR = Path(__file__).parent.parent.parent.parent.joinpath("config").resolve()
 
-# Read JSON files
-ZONES_CONFIG = json.load(open(CONFIG_DIR.joinpath("zones.json"), encoding="utf-8"))
-EXCHANGES_CONFIG = json.load(
-    open(CONFIG_DIR.joinpath("exchanges.json"), encoding="utf-8")
-)
-CO2EQ_PARAMETERS_ALL = json.load(
-    open(CONFIG_DIR.joinpath("co2eq_parameters_all.json"), encoding="utf-8")
-)
-CO2EQ_PARAMETERS_LIFECYCLE = {
-    **CO2EQ_PARAMETERS_ALL,
-    **json.load(
-        open(CONFIG_DIR.joinpath("co2eq_parameters_lifecycle.json"), encoding="utf-8")
-    ),
+defaults = yaml.safe_load(open(CONFIG_DIR.joinpath("defaults.yaml"), encoding="utf-8"))
+
+zones_config = {}
+for zone_path in CONFIG_DIR.joinpath("zones").glob("*.yaml"):
+    zone_key = zone_path.stem
+    zones_config[zone_key] = yaml.safe_load(open(zone_path, encoding="utf-8"))
+
+exchanges_config = {}
+for exchange_path in CONFIG_DIR.joinpath("exchanges").glob("*.yaml"):
+    _exchange_key_unicode = exchange_path.stem
+    _zone_keys = _exchange_key_unicode.split("→")
+    assert len(_zone_keys) == 2
+    exchange_key = "->".join(_zone_keys)
+    exchanges_config[exchange_key] = yaml.safe_load(
+        open(exchange_path, encoding="utf-8")
+    )
+
+
+co2eq_parameters_all = {
+    k: {
+        "defaults": defaults[k],
+        "zoneOverrides": {},
+    }
+    for k in ["fallbackZoneMixes", "isLowCarbon", "isRenewable"]
 }
-CO2EQ_PARAMETERS_DIRECT = {
-    **CO2EQ_PARAMETERS_ALL,
-    **json.load(
-        open(CONFIG_DIR.joinpath("co2eq_parameters_direct.json"), encoding="utf-8")
-    ),
+co2eq_parameters_direct = {
+    "emissionFactors": {
+        "defaults": defaults["emissionFactors"]["direct"],
+        "zoneOverrides": {},
+    },
 }
+co2eq_parameters_lifecycle = {
+    "emissionFactors": {
+        "defaults": defaults["emissionFactors"]["lifecycle"],
+        "zoneOverrides": {},
+    },
+}
+# Populate zone overrides
+for zone_key, zone_config in zones_config.items():
+    for k in ["fallbackZoneMixes", "isLowCarbon", "isRenewable"]:
+        if k in zone_config:
+            co2eq_parameters_all[k]["zoneOverrides"][zone_key] = zone_config[k]
+            del zone_config[k]
+    if "emissionFactors" in zone_config:
+        for k in ["direct", "lifecycle"]:
+            if k in zone_config["emissionFactors"]:
+                if k == "direct":
+                    co2eq_parameters_direct["emissionFactors"]["zoneOverrides"][
+                        zone_key
+                    ] = zone_config["emissionFactors"][k]
+                elif k == "lifecycle":
+                    co2eq_parameters_lifecycle["emissionFactors"]["zoneOverrides"][
+                        zone_key
+                    ] = zone_config["emissionFactors"][k]
+        del zone_config["emissionFactors"]
```

Comment:
@pierresegonne is there an important reason to delete the `emissionFactors` key from the zone configs after they're loaded?

As context: I'm looking into writing a test/validator that asserts on the equality of the `sources` values between the top-level zone config and individual emissions factor entries, and that caused me to notice that this key gets removed.

Reply:
Hey, 
It's purely for memory reasons. In the backend we use the `co2eq_parameters_**` config objects to interact with the emission factors, not the `zone_config`, so we remove that field to reduce clutter. 
That could be challenged :)